website_sale_menu_top_user_selling/controllers/main.py

```diff
@@ -0,0 +1,90 @@
+from odoo import http
+from odoo.http import request
+
+from odoo.addons.website_sale.controllers.main import TableCompute, WebsiteSale
+
+
+class WebsiteSale(WebsiteSale):
+    @http.route(
+        ["/shop/top_selling_products", "/shop/top_selling_products/page/<int:page>"],
+        type="http",
+        auth="public",
+        website=True,
+    )
+    def user_top_products(self, page=0, ppg=False, **kwargs):
+        if request.env.user.has_group("base.group_public"):
+            return request.redirect("/web/login")
+        param_limit = int(
+            request.env["ir.config_parameter"]
+            .sudo()
+            .get_param("website_sale_menu_top_user_selling.limit", 10)
+        )
+        # Get best-selling products from the user
+        product_data = (
+            request.env["sale.order.line"]
+            .sudo()
+            .read_group(
+                [("order_id.partner_id", "=", request.env.user.partner_id.id)],
```

Comment:
Maybe you should restrict the website or at least the company..

Reply:
Better to limit both and leave no loose ends. The problem is that they were not well declared in the manifest, haste is not good. Actually, whether they are published or not is not a problem since they are running as administrator and will appear shaded.